.github/workflows/gchr_app.yml

```diff
@@ -0,0 +1,150 @@
+# Workflow that builds, tests and then pushes the docker images to the ghcr.io repository
+name: Build and Publish App Image
+
+# Only run one workflow of the same group at a time.
+# There can be at most one running and one pending job in a concurrency group at any time.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+on:
```

Comment:
Add comments:
# Always run on "main"
# Always run on tags
# Always run on PRs
# Can also be triggered manually